mesop/components/table/table.py

```diff
@@ -0,0 +1,68 @@
+from dataclasses import dataclass
+from typing import Any, Protocol, runtime_checkable
+
+import mesop.components.table.table_pb2 as table_pb
+from mesop.component_helpers import insert_component, register_native_component
+
+
+@runtime_checkable
+class TableDataSource(Protocol):
+  """Interface for supporting different table data sources"""
+
+  def columns(self) -> list[str]:
+    """Returns a list of columns in the data source."""
+    pass
+
+  def to_data_source(self) -> list[table_pb.TableRow]:
+    """Returns the data source to match table_pb.TableType."""
+    pass
+
+
+@dataclass
+class TableDataSourcePandasDataFrame(TableDataSource):
+  # TODO: Need a Pydantic validator for pandas DataFrames. Temporarily using Any for now.
+  df: Any
+
+  def columns(self) -> list[str]:
+    return list(self.df.columns)
+
+  def to_data_source(self) -> list[table_pb.TableRow]:
+    columns = list(self.df.columns)
+    data = []
+    for df_row in self.df.itertuples():
+      row = {}
+      # TODO: Properly support the various Pandas data types.
+      for col_name in columns:
+        value = getattr(df_row, col_name)
+        if isinstance(value, bool):
+          row[col_name] = table_pb.TableValue(bool_value=value)
+        elif isinstance(value, str):
+          row[col_name] = table_pb.TableValue(string_value=value)
+        elif isinstance(value, int):
+          row[col_name] = table_pb.TableValue(int_value=value)
+        elif isinstance(value, float):
+          row[col_name] = table_pb.TableValue(float_value=value)
+        else:
+          raise ValueError(f"Unsupported Mesop table data type: {type(value)}")
+      data.append(table_pb.TableRow(row=row))
+    return data
+
+
+@register_native_component
+def table(data_source: TableDataSourcePandasDataFrame):
```

Comment:
The parameter should just take in a Pandas DataFrame, which would be a much cleaner callsite: `me.table(df)`. We can wrap it internally with a dataclass if needed.

Reply:
Sure. That sounds fine. I mainly did this initially because of the Pydantic type checks. Thought I could trick it into working, but it didn't work. So I'll updated to just pass in the dataframe.